examples/siro_sandbox/sandbox_app.py

```diff
@@ -636,35 +711,134 @@ def sim_update(self, dt):
         # 2) press left mouse button and move mouse
         self._camera_pitch_and_yaw_mouse_control()
 
-        agent_idx = self.ctrl_helper.get_gui_controlled_agent_index()
-        if agent_idx is None:
-            self._free_camera_lookat_control()
-            lookat = self.lookat
-        else:
-            art_obj = (
-                self.get_sim().agents_mgr[agent_idx].articulated_agent.sim_obj
+        lookat = self._create_camera_lookat()
+        self.cam_transform = mn.Matrix4.look_at(
+            lookat[0], lookat[1], mn.Vector3(0, 1, 0)
+        )
+
+    def _generate_tutorial(self) -> List[TutorialStage]:
```

Comment:
This looks like a lot of logic. Can we move this to a separate helper class and file? `SandboxDriver` is becoming bloated.